File: backend/app/services/targets.py
ACTIVITY = {
    "sedentary": 1.2,
    "light": 1.375,
    "moderate": 1.55,
    "active": 1.725,
    "very_active": 1.9,
}
GOAL_DELTA = {"lose": -500, "maintain": 0, "gain": 300}


def bmr(sex: str, weight_kg: float, height_cm: float, age: int) -> float:
    base = 10 * weight_kg + 6.25 * height_cm - 5 * age
    return base + (5 if sex == "male" else -161)


def daily_calories(sex: str, weight_kg: float, height_cm: float, age: int, activity_level: str, goal: str) -> int:
    tdee = bmr(sex, weight_kg, height_cm, age) * ACTIVITY[activity_level]
    return round(tdee + GOAL_DELTA[goal])


def macro_targets(daily_cal: int) -> dict[str, int]:
    protein_g = round((daily_cal * 0.30) / 4)
    carbs_g = round((daily_cal * 0.40) / 4)
    fat_g = round((daily_cal * 0.30) / 9)
    return {
        "daily_calorie_target": daily_cal,
        "daily_protein_target_g": protein_g,
        "daily_carbs_target_g": carbs_g,
        "daily_fat_target_g": fat_g,
    }


def compute_targets(
    sex: str,
    weight_kg: float,
    height_cm: float,
    age: int,
    activity_level: str,
    goal: str,
) -> dict[str, int]:
    cal = daily_calories(sex, weight_kg, height_cm, age, activity_level, goal)
    return macro_targets(cal)

File: mobile/lib/targets.ts
const ACTIVITY: Record<string, number> = {
  sedentary: 1.2,
  light: 1.375,
  moderate: 1.55,
  active: 1.725,
  very_active: 1.9,
};
const GOAL_DELTA: Record<string, number> = { lose: -500, maintain: 0, gain: 300 };

export function bmr(sex: "male" | "female", weightKg: number, heightCm: number, age: number) {
  const base = 10 * weightKg + 6.25 * heightCm - 5 * age;
  return base + (sex === "male" ? 5 : -161);
}

export function computeTargets(
  sex: "male" | "female",
  weightKg: number,
  heightCm: number,
  age: number,
  activity: string,
  goal: string
) {
  const cal = Math.round(bmr(sex, weightKg, heightCm, age) * ACTIVITY[activity] + GOAL_DELTA[goal]);
  return {
    daily_calorie_target: cal,
    daily_protein_target_g: Math.round((cal * 0.3) / 4),
    daily_carbs_target_g: Math.round((cal * 0.4) / 4),
    daily_fat_target_g: Math.round((cal * 0.3) / 9),
  };
}

export function lbsToKg(lbs: number) {
  return lbs / 2.205;
}
export function kgToLbs(kg: number) {
  return kg * 2.205;
}
export function inToCm(inches: number) {
  return inches * 2.54;
}
export function cmToIn(cm: number) {
  return cm / 2.54;
}

export function roundCal(n: number) {
  return Math.round(n / 5) * 5;
}
export function roundG(n: number) {
  return Math.round(n);
}
